scripts/quests/ahtUrhgan/Promotion_Chief_Sergeant.lua

```diff
@@ -0,0 +1,293 @@
+-----------------------------------
+-- Promotion: Chief Sergeant
+-- Logid: 6 Questid: 96
+-- Abquhbah: !pos 35 -6 -58 50
+-- Hagakoff: !pos 82 0 -74 50
+-- Mushroom Patch:
+--[[
+    [1] = !pos 123 37 6 68
+    [2] = !pos -359 15 -425 68
+    [3] = !pos -208 9 143 68
+    [4] = !pos 117 37 -46 68
+    [5] = !pos -221 9 141 68
+    [6] = !pos -345 15 -442 68
+]]
+-----------------------------------
+local aydeewaID = zones[xi.zone.AYDEEWA_SUBTERRANE]
+local whitegateID = zones[xi.zone.AHT_URHGAN_WHITEGATE]
+-----------------------------------
+
+local quest = Quest:new(xi.questLog.AHT_URHGAN, xi.quest.id.ahtUrhgan.PROMOTION_CHIEF_SERGEANT)
+local patchStage =
+{
+    NONE      = 0,
+    HARVESTED = 1,
+    PLANTED   = 2,
+    PICKED    = 3,
+}
+local triggerPatch = function(player, npc)
+    --[[
+    notes: dont have KI and havent plated, recieve KI - cs 14
+    place you have picked from - cs 15
+    have KI and wasnt place you picked it, lose KI - 19, option 0
+    place you have plated but not ready yet - cs 20
+    place you placed and ready to harvest - cs 21 {[0] = amount, [3] = bonus mushroom} option = amount recieved
+    if no KI and already harvested the area - cs 18
+    have KI and cant get KI from - cs 16
+    ]]
+
+    local mushroomPatchOffset = npc:getID() - aydeewaID.npc.MUSHROOM_PATCH
+    local patch = 'MPatch'..mushroomPatchOffset
+    local mushroomVar = quest:getVar(player, patch)
+    local result = quest:messageSpecial(aydeewaID.text.NOTHING_OUT_OF_ORDINARY)
+    local time = VanadielHour()
+
+    -- Dont have KI have not picked or planted at patch
+    if
+        not player:hasKeyItem(xi.ki.SCOURSHROOM) and
+        mushroomVar == patchStage.NONE
+    then
+        return quest:progressEvent(14)
+
+    -- Section you have already Picked from
+    elseif mushroomVar == patchStage.HARVESTED then
+        return quest:progressEvent(15)
+
+    -- Section Planting if have KI and has not yet been touched
+    elseif
+        player:hasKeyItem(xi.ki.SCOURSHROOM) and
+        mushroomVar == patchStage.NONE
+    then
+        return quest:progressEvent(19)
+
+    -- planted a mushroom
+    elseif mushroomVar == patchStage.PLANTED then
+        -- Ready to harvest
+        if
+            quest:getVar(player, 'MPatch_Time') <= VanadielUniqueDay() and
+            not player:needToZone()
+        then
+            local timeBonus = 1
+            local bonus = math.random(100) > 75 and 5 or 0
+            if time > 4 and time < 6 and math.random(100) > 50 then
+                timeBonus = 2
+            end
```

Comment:
`local time = VanadielHour()` will return 1, 2, 3, 4, etc...
So you could just check for `time == 5`

As for the random, could you define the minimum aswell?
`if time == 5 and math.random(1, 100) <= 50 then`

Reply:
this means it can harvest at hours 4-6 not just at 5 but i do need to add the >= <= for inclusion, so still a good catch to make me check over what i was doing